agent/directory.go

```diff
@@ -300,11 +301,13 @@ func (a *Agent) SetHomeDirectory() {
 		return
 	}
 
-	usr, err := user.Current()
+	userHome, err := homedir.Dir()
 	if err != nil {
-		grip.Warning(errors.Wrap(err, "getting current user to set the home directory"))
-		return
+		// workaround for cygwin if we're on windows but couldn't get a homedir
+		if runtime.GOOS == "windows" && len(os.Getenv("HOME")) > 0 {
+			userHome = os.Getenv("HOME")
+		}
```

Comment:
Should we still return early here if it isn't windows / the os.Getenv("HOME")'s length is 0?

Also, could you reword the comment to be more general? It seems difficult to interpret in 3+ months. Maybe like "On Windows hosts using cygwin, we have to fallback to the HOME env variable. This is because these hosts will set a HOME directory variable that should be used as a reference".

Reply:
This bit of logic was being reused both here and in the cli so I extracted it to it's own function and returned the error better. I don't think it should return early though because we should still attempt to use it to find the files. We have enough error checking on the file paths to fail gracefully and whatever it ends up returning may be better than an empty string. 

I updated the comment to "On Windows hosts using cygwin, we have to fallback to the HOME env variable.  ", but I don't know if "This is because these hosts will set a HOME directory variable that should be used as a reference." is necessary because the code that follows shows that. I like to keep comments to only context that can't be gleaned from the code itself.